web-module/src/main/webapp/resources/structEditJS/dateTime/celements-date-time-field.js

```diff
@@ -162,317 +162,352 @@
-        const _me = this;
-        const pickerConfigObj = Object.assign({
-          'allowBlank': true,
-          'dayOfWeekStart': 1,
-          'format': 'd.m.Y',
-          'timepicker': false
-        }, configObj);
-        return new CELEMENTS.structEdit.DateOrTimeFieldPicker(dateInputField, '.CelDatePicker',
-          "dd.MM.y", pickerConfigObj, _me._dateFieldValidator);
-      },
+  class CelementsDateTimePickerFactory {
+
+    createDatePickerField(dateInputField, configObj = {}) {
+      const pickerConfigObj = Object.assign({
+        'allowBlank': true,
+        'dayOfWeekStart': 1,
+        'format': 'd.m.Y',
+        'timepicker': false
+      }, configObj);
+      return new CELEMENTS.structEdit.DateOrTimeFieldPicker(dateInputField,
+        '.CelDatePicker', 'dd.MM.y', pickerConfigObj, this.#dateFieldValidator);
+    }
 
-      _dateFieldValidator: function(value) {
-        console.debug("dateFieldValidator - from", value);
-        value = (value || "").toString().trim().replace(/[,-]/g, '.');
-        const split = value.split(".")
-          .filter(function(elem) { return elem; }); // filter falsy elements
-        const day = Number(split[0]);
-        const month = Number(split[1]);
-        let year = Number(split[2]);
-        if (year < 100) {
-          year += Math.floor(new Date().getFullYear() / 100) * 100; // 21 -> 2021
-        }
-        let validated = "";
-        if (value
-          && (!split[0] || (!isNaN(day) && (day > 0) && (day <= 31)))
-          && (!split[1] || (!isNaN(month) && (month > 0) && (month <= 12)))
-          && (!split[2] || (!isNaN(year) && (year > 100) && (year <= 9999)))) {
-          const curDate = new Date();
-          const date = new Date(year || curDate.getFullYear(), (month || (curDate.getMonth() + 1)) - 1,
-            day || curDate.getDate());
-          validated = $j.format.date(date, "dd.MM.y");
-        }
-        console.debug("dateFieldValidator - to", validated);
-        return validated;
-      },
+    #dateFieldValidator(value) {
+      console.debug("dateFieldValidator - from", value);
+      value = (value || '').toString().trim().replace(/[,-]/g, '.');
+      const split = value.split('.').filter(Boolean);
+      const day = Number(split[0]);
+      const month = Number(split[1]);
+      let year = Number(split[2]);
+      if (year < 100) {
+        year += Math.floor(new Date().getFullYear() / 100) * 100; // 21 -> 2021
+      }
+      let validated = '';
+      if (value
+        && (!split[0] || (!isNaN(day) && (day > 0) && (day <= 31)))
+        && (!split[1] || (!isNaN(month) && (month > 0) && (month <= 12)))
+        && (!split[2] || (!isNaN(year) && (year > 100) && (year <= 9999)))) {
+        const curDate = new Date();
+        const date = new Date(year || curDate.getFullYear(), (month || (curDate.getMonth() + 1)) - 1,
+          day || curDate.getDate());
+        validated = $j.format.date(date, 'dd.MM.y');

```

Comment:
Please replace `$j.format.date` with vanilla js replacement:
```suggestion
        validated = new Intl.DateTimeFormat('de-CH', {day : '2-digit', month : '2-digit', year : 'numeric' }).format(date);

```

Reply:
Nice find, I didn't find out about `Intl.DateTimeFormat` despite doing some googling. Though two observations:
- It doesn't support parsing strings to dates
- It uses quite complex formatting rules (instead of `dd.MM.yyyy`) which could make it more difficult to configure in the future

Reply:
for parsing I suggest to use split+newDate, Luxon.js or hopefully Temporal will be in place sooner than later:
- Luxon.js: https://moment.github.io/luxon/
- Temporal: https://momentjs.com/docs/#/-project-status/future/